src/pages/Lending/Pool/components/Repay/RepayInput.tsx

```diff
@@ -731,7 +732,8 @@ export const RepayInput = ({
           label={translate(quoteErrorTooltipTranslation)}
           isDisabled={!lendingQuoteCloseError}
         >
-          <Button
+          <ButtonWalletPredicate
+            isValidWallet={!isDemoWallet}
```

Comment:
q: does it mean that any wallet that's not native demo will be considered a "valid wallet", even if it doesn't support the chain?

Reply:
Actually yep, but I feel like we might need another wording if we want to add other cases because don't you think that it looks weird to display `Connect Wallet` while you already have a connected wallet? I would expect something like `Connect a supported wallet` or `Connect another wallet` 

Happy to discuss it and makes a lot of sense

Reply:
As discussed, I did move the demo wallet logic to the abstraction and forced the use of `isValidWallet` in [8fc6ffb](https://github.com/shapeshift/web/pull/7270/commits/8fc6ffbce7af6fb268266cf75b5c573428cf4653)